app/utils/JsonFile.scala

```diff
@@ -16,19 +16,44 @@
 
 package utils
 
-import play.api.libs.json.{JsArray, JsString, Json}
+import play.api.libs.json.{JsArray, JsString, Json, Reads}
+
+import scala.util.{Failure, Success, Try}
 
 object JsonFile {
   def readFromJsonFile[T](file: String, deserializer: (String, String) => T): List[T] = {
-    val jsonFile = getClass.getResourceAsStream(file)
+    val jsonInputStream = getClass.getResourceAsStream(file)
 
-    Json.parse(jsonFile) match {
+    Json.parse(jsonInputStream) match {
       case JsArray(cs) =>
         cs.toList.collect {
           case JsArray(Seq(label: JsString, code: JsString)) =>
             deserializer(label.value, code.value)
         }
-      case _ => throw new IllegalArgumentException("Could not read JSON array from : " + jsonFile)
+      case _ => throwError(file)
     }
   }
+
+  def getJsonArrayFromFile[T](file: String, reader: Reads[T]): List[T] = {
+    val jsonInputStream = getClass.getResourceAsStream(file)
+
+    Try(Json.parse(jsonInputStream)) match {
+      case Success(JsArray(jsValues)) =>
+        val items = jsValues.toList.map { jsValue =>
+          reader.reads(jsValue).asOpt
+        }
+
+        if (items.contains(None)) {
+          throw new IllegalArgumentException(s"One or more entries in JSON file: '$file' could not be parsed!")
```

Comment:
The error message is not clear (but it's just me :smile:). Why cannot the file be parsed?

Reply:
Well each element in the array needs to be parsed and each element could have a different parsing issue. I did not think it worth while to try and collect and list every pare error as its an 'all or nothing' operation (either the whole file can be parsed or none of it will be taken).

Reply:
tbh I was just asking to make the error message a bit more readable. Something like
`'$file' could not be parsed due to one or more invalid JSON attributes` or similar